### PR TITLE
Add ability to create Database Clusters

### LINKED ISSUE #9: Feature: Add ability to create database clusters
### What problem are you facing?

I would like the ability to create a Database Cluster using resource definitions with Crossplane.

You can view the API reference [here](https://docs.digitalocean.com/reference/api/api-reference/#operation/create_database_cluster)

### PR DESCRIPTION
**Pull request by ADustyOldMuffin**
### Description of your changes

This adds the ability to specify a Database Cluster and deploy to DigitalOcean.

Fixes #9 

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

I have Crossplane installed and added the CRD to a `kind` cluster with my credentials setup. I monitored the deployment and saw all of the observed state update with the deployment.

### CODE REVIEW

**Comment by ADustyOldMuffin** [comment]
Re-created in an attempt to get no-op to leave it alone.

**Comment by ADustyOldMuffin** [comment]
So, I have no idea what the issue is with `check-diff`, I've ran it multiple time, checked it out fresh and built, and still nothing.

File: examples/database/database.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  forProvider:
+    engine: pg
+    version: "13"
+    numNodes: 3
+    size: db-s-2vcpu-4gb
+    region: nyc3
+    tags:
+      - "from-crossplane"
+  providerConfigRef:
+    name: example-config
```
Comment (kimschles) [inline examples/database/database.yaml:15]: Line 15 needs to be changed to `name: example` to work with the example provider.

**Review by kimschles — CHANGES_REQUESTED** [review]
Hi @ADustyOldMuffin! 

I tested out this branch, and I was able to create a DO postgres cluster! :tada:

The only issue is that the providerConfigRef name needs to be changed from `example-config` to `example`. 

Once that's changed, I can approve and you can merge the PR. Great work!

File: examples/database/database.yaml (hunk 1 above)
Comment (ADustyOldMuffin) [reply examples/database/database.yaml]: finished, thanks!

**Review by kimschles — APPROVED** [review]

**Review by hasheddan — COMMENTED** [review]
@ADustyOldMuffin e2e tests should pass if you change:
https://github.com/crossplane-contrib/provider-digitalocean/blob/2ce98ea49b08f96def8c5d9ddb3470295dd6f006/cluster/local/integration_tests.sh#L155

from `v1beta1` to `v1` 👍🏻 

The `v1beta1` version was recently dropped after deprecation period in Crossplane.

**Pull request merged**